### PR TITLE
net-dns/knot-resolver: add 5.7.4 with fixed redirects

### PR DESCRIPTION
**Pull request by fkobi**
---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: net-dns/knot-resolver/knot-resolver-5.7.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+EAPI=8
+
+LUA_COMPAT=( luajit )
+
+inherit lua-single meson tmpfiles verify-sig
+
+DESCRIPTION="A scaleable caching DNS resolver"
+HOMEPAGE="https://www.knot-resolver.cz https://gitlab.nic.cz/knot/knot-resolver"
+SRC_URI="
+	https://knot-resolver.nic.cz/release/${P}.tar.xz
```
Comment (juippis) [inline net-dns/knot-resolver/knot-resolver-5.7.4.ebuild:13]:
```
>>> Downloading 'https://knot-resolver.nic.cz/release/knot-resolver-5.7.4.tar.xz'
--2024-09-23 12:30:56--  https://knot-resolver.nic.cz/release/knot-resolver-5.7.4.tar.xz
Resolving knot-resolver.nic.cz... 217.31.202.46, 2001:1488:ffff::46
Connecting to knot-resolver.nic.cz|217.31.202.46|:443... connected.
ERROR: no certificate subject alternative name matches
        requested host name 'knot-resolver.nic.cz'.
To connect to knot-resolver.nic.cz insecurely, use `--no-check-certificate'.
!!! Couldn't download 'knot-resolver-5.7.4.tar.xz'. Aborting.
```
Although I can't even download https://secure.nic.cz/files/knot-resolver/knot-resolver-5.7.4.tar.xz from their official web page, I need to try later.

File: net-dns/knot-resolver/knot-resolver-5.7.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	"${FILESDIR}"/${PN}-5.7.0-r2-tmpfiles.patch
+)
+
+VERIFY_SIG_OPENPGP_KEY_PATH=/usr/share/openpgp-keys/${PN}.gpg
+
+src_unpack() {
+	use verify-sig && verify-sig_verify_detached "${DISTDIR}"/${P}.tar.xz{,.asc}
+
+	unpack ${P}.tar.xz
+}
```
Comment (juippis) [inline net-dns/knot-resolver/knot-resolver-5.7.4.ebuild:66]: I think this is how the eclass works, should be able to remove this phase...?

File: net-dns/knot-resolver/knot-resolver-5.7.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		-Ddoc=disabled
+		-Ddocdir="${EPREFIX}"/usr/share/doc/${PF}
+		-Dopenssl=disabled
+		-Dmalloc=$(usex jemalloc jemalloc disabled)
+		$(meson_feature caps capng)
+		$(meson_feature dnstap)
+		$(meson_feature kresc client)
+		$(meson_feature nghttp2)
+		$(meson_feature test unit_tests)
+		$(meson_feature systemd systemd_files)
```
Comment (juippis) [inline net-dns/knot-resolver/knot-resolver-5.7.4.ebuild:82]: Does this only install the service files based on this use flag? They should always be installed. If it doesn't link to anything systemd-related, this use flag can be removed and always install the service files unconditionally.

File: net-dns/knot-resolver/knot-resolver-5.7.4.ebuild (hunk 2 above)
Comment (PPN-SD) [reply net-dns/knot-resolver/knot-resolver-5.7.4.ebuild:66]: Yes. It works without.
```suggestion
```

File: net-dns/knot-resolver/knot-resolver-5.7.4.ebuild (hunk 3 above)
Comment (PPN-SD) [reply net-dns/knot-resolver/knot-resolver-5.7.4.ebuild:82]: systemd is actually required for something other than systemd_files (systemd watchdog). I suggest a patch for that. See the other comment.

```suggestion
		-Dsystemd_files=enabled
		$(meson_feature systemd)
```

File: net-dns/knot-resolver/knot-resolver-5.7.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	meson_src_install
+	fowners -R ${PN}: /etc/${PN}
+
+	newinitd "${FILESDIR}"/kresd.initd-r1 kresd
+	newconfd "${FILESDIR}"/kresd.confd-r1 kresd
+}
+
+pkg_postinst() {
+	tmpfiles_process knot-resolver.conf
+}
```
Comment (PPN-SD) [inline net-dns/knot-resolver/knot-resolver-5.7.4.ebuild:98]:
```suggestion
```

File: net-dns/knot-resolver/knot-resolver-5.7.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	virtual/pkgconfig
+	verify-sig? ( >=sec-keys/openpgp-keys-knot-resolver-20240304 )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-5.5.3-docdir.patch
+	"${FILESDIR}"/${PN}-5.5.3-nghttp-openssl.patch
+
+	# Bug #921567
+	"${FILESDIR}"/${PN}-5.7.0-r2-tmpfiles.patch
```
Comment (PPN-SD) [inline net-dns/knot-resolver/knot-resolver-5.7.4.ebuild:57]: I suggest a patch for systemd ([knot-resolver-5.7.4-libsystemd.txt](https://github.com/user-attachments/files/17335941/knot-resolver-5.7.4-libsystemd.txt) .patch don't upload), remove this one and force systemd_files instead.

```suggestion
	"${FILESDIR}"/${PN}-5.7.4-libsystemd.patch
```

**Review by PPN-SD — COMMENTED** [review]
Force systemd_files. systemd useflag is used only for libsystemd dependency.

File: net-dns/knot-resolver/knot-resolver-5.7.4.ebuild (hunk 2 above)
Comment (PPN-SD) [reply net-dns/knot-resolver/knot-resolver-5.7.4.ebuild:66]: Yes. It works without.
```suggestion
```

File: net-dns/knot-resolver/knot-resolver-5.7.4.ebuild (hunk 3 above)
Comment (PPN-SD) [reply net-dns/knot-resolver/knot-resolver-5.7.4.ebuild:82]: Suggest use this useflag for libsystemd only and force systemd_files : see review below.

```suggestion
		-Dsystemd_files=enabled
		$(meson_feature systemd)
```

**Comment by PPN-SD** [comment]
@juippis, do you think I could do a new PR with fixed redirects and systemd changes ?

**Comment by PPN-SD** [comment]
> @PPN-SD I see have lost some of my free time lately. IF you wish to take over the changes and file them in your PR then let me know so I can close this.

Ok to close and new PR.

**Comment by PPN-SD** [comment]
@fkobi could you close to avoid duplicate ?

**Pull request closed without merging**